tmt/steps/__init__.py

```diff
@@ -1958,7 +1958,16 @@ def _login(
             # Attempt to push the workdir to the guest
             try:
                 guest.push()
-                cwd = cwd or self.parent.plan.worktree
+                if not cwd:
+                    worktree = self.parent.plan.worktree
+                    tests = self.parent.plan.discover.tests()
+                    test_path = tests[-1].path if tests else None
+
+                    if worktree and test_path:
```

Comment:
I wouldn't even check for the worktree variable, or alternatively you should include that condition on the else branch too

Reply:
I needed to check it as I am trying to get `.parent` content from it and without that I would get mypy error
So in case if worktree is None, then I would raise exception.

Reply:
yeah, I thought it would be like that, but don't really like not checking it again on the else branch